Toolkit API docs

Loquacity opened the pull request:
# Description

Adds API docs for hyperfunctions, and reorganizes according to plan in issue.

# Version

Which documentation version does this PR apply to?

- [x] Latest (Default)
- [ ] Version 1.7
- [ ] Older [specify]

# Links

Partial https://github.com/timescale/docs-private/issues/4
Related PR: File setup to rename section and move install doc in timescale/docs#504

Loquacity commented:
![image](https://user-images.githubusercontent.com/3914967/137274033-fbb072c6-4233-42b9-8323-ebae2402e46e.png)
![image](https://user-images.githubusercontent.com/3914967/137274086-dfc08bec-29d7-4a62-a504-3819e3ab90ef.png)
![image](https://user-images.githubusercontent.com/3914967/137274120-5538e3e7-cb77-4045-b302-b4f5ceb5c3fa.png)
![image](https://user-images.githubusercontent.com/3914967/137274178-1a995763-85cb-494c-bff3-db60b436056a.png)
![image](https://user-images.githubusercontent.com/3914967/137274231-af9abf60-4803-42e6-b50a-c3977634ff4c.png)
![image](https://user-images.githubusercontent.com/3914967/137274280-ef3a5fe2-f122-4fa2-bd62-c2c76403e97a.png)

Loquacity commented:
💥  Now with better intros! 💥 

![image](https://user-images.githubusercontent.com/3914967/137423368-3b7a92b9-3b58-4d1f-b819-df825297785e.png)
![image](https://user-images.githubusercontent.com/3914967/137423414-49387c8e-64ea-4876-a0ee-4f8173c19449.png)
![image](https://user-images.githubusercontent.com/3914967/137423460-ab448a97-3b76-4139-8373-addf9fa2e9d9.png)
![image](https://user-images.githubusercontent.com/3914967/137423488-899efaa5-d7dc-4ba6-83f9-72931d8a3723.png)
![image](https://user-images.githubusercontent.com/3914967/137423518-09fe068b-0ecc-471e-998a-625ae088876e.png)
![image](https://user-images.githubusercontent.com/3914967/137423556-0bcab168-209f-4dd6-b889-4ecf6bc37770.png)
![image](https://user-images.githubusercontent.com/3914967/137423594-d656ae24-a86e-407d-9a06-1919fd8931a9.png)

davidkohn88 reviewed (approved):
These look pretty close, there's a few things I want to add, namely table of contents with links to each thing inside a function family at each top level function family api...also we need stats_agg docs. I can work on that tomorrow and at least get something together...but this is already a huge PR, I think getting this in and then making PRs against our main branch is a good way forward and this at least establishes the main structure. 

The only thing I'd also like to mention is that the things that we've removed, the advanced guides and stuff, we should probably go back and either add redirects or make sure that all the links in old blogs etc are properly updated...

Loquacity commented:
> These look pretty close, there's a few things I want to add, namely table of contents with links to each thing inside a function family at each top level function family api

Added the portions of the toolkit matrix to the function family pages, I hope that satisfies the requirement? I'll go back through and add links to the pages as well, in the next pass.

> also we need stats_agg docs. I can work on that tomorrow and at least get something together.

OK 👍 

> The only thing I'd also like to mention is that the things that we've removed, the advanced guides and stuff, we should probably go back and either add redirects or make sure that all the links in old blogs etc are properly updated...

I'll do another pass through looking for these.

Loquacity commented:
@davidkohn88 Happy for you to merge this at will! 🚀
